specification/migrate/resource-manager/Microsoft.Migrate/stable/2019-05-01/migrate.json

```diff
@@ -0,0 +1,3778 @@
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
+        },
+        "message": {
+          "type": "string",
+          "description": "A message describing the error, intended to be suitable for display in a user interface."
+        },
+        "target": {
+          "type": "string",
+          "description": "The target of the particular error. For example, the name of the property in error."
+        },
+        "details": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          },
+          "description": "A list of additional details about the error."
+        }
+      },
+      "description": "An error response from the Azure Migrate service."
+    },
+    "CollectorBodyAgentSpnProperties": {
+      "type": "object",
+      "properties": {
+        "authority": {
+          "description": "AAD Authority URL which was used to request the token for the service principal.",
+          "type": "string"
+        },
+        "applicationId": {
+          "description": "Application/client Id for the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        },
+        "audience": {
+          "description": "Intended audience for the service principal.",
+          "type": "string"
+        },
+        "objectId": {
+          "description": "Object Id of the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "Tenant Id for the service principal with which the on-premise management/data plane components would communicate with our Azure services.",
+          "type": "string"
+        }
+      }
+    },
+    "HyperVCollector": {
+      "type": "object",
+      "properties": {
+        "eTag": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "readOnly": true
+        },
+        "properties": {
+          "$ref": "#/definitions/CollectorProperties"
+        }
+      }
+    },
+    "CollectorProperties": {
+      "type": "object",
+      "properties": {
+        "discoverySiteId": {
+          "type": "string"
+        },
+        "createdTimestamp": {
+          "type": "string",
+          "readOnly": true
+        },
+        "updatedTimestamp": {
+          "type": "string",
+          "readOnly": true
+        },
+        "agentProperties": {
+          "$ref": "#/definitions/CollectorAgentProperties"
+        }
+      }
+    },
+    "CollectorAgentProperties": {
+      "type": "object",
+      "properties": {
+        "id": {
```

Comment:
Having an "id" property inside of properties gets confusing since the resource itself has a top-level "id".  This should be named something else if possible